spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesInformerDiscoveryClient.java

```diff
@@ -89,13 +90,32 @@ public KubernetesInformerDiscoveryClient(SharedInformerFactory sharedInformerFac
 			Lister<V1Service> serviceLister, Lister<V1Endpoints> endpointsLister,
 			SharedInformer<V1Service> serviceInformer, SharedInformer<V1Endpoints> endpointsInformer,
 			KubernetesDiscoveryProperties properties) {
-		this.sharedInformerFactory = sharedInformerFactory;
-		this.serviceLister = serviceLister;
-		this.endpointsLister = endpointsLister;
+		this.sharedInformerFactories = List.of(sharedInformerFactory);
+		this.serviceListers = List.of(serviceLister);
+		this.endpointsListers = List.of(endpointsLister);
 		this.informersReadyFunc = () -> serviceInformer.hasSynced() && endpointsInformer.hasSynced();
 		this.properties = properties;
 	}
 
+	public KubernetesInformerDiscoveryClient(List<SharedInformerFactory> sharedInformerFactories,
```

Comment:
a new constructor that is only called when `spring.cloud.kubernetes.discovery.namespaces` is present, that only accepts "Lists" of Lister/SharedInformer/SharedInformerFactory.